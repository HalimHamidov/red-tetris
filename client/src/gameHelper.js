import { Tetrominos } from "./tetrominos";
export const S_WIDTH = 10;
export const S_HEIGHT = 20;


export const Createstage = (h = S_HEIGHT, w = S_WIDTH) =>
  Array.from(Array(h), () => new Array(w).fill([0, "clear"]));

export const checkcollision = (player, stage, { x: moveX, y: moveY }) => {
  for (let y = 0; y < player.tetromino.length; y++) {
    for (let x = 0; x < player.tetromino.length; x++) {
      if (player.tetromino[y][x] !== 0) {
        if (
          !stage[y + player.pos.y + moveY] ||
          !stage[y + player.pos.y + moveY][x + player.pos.x + moveX] ||
          stage[y + player.pos.y + moveY][x + player.pos.x + moveX][1] !==
            "clear"
        )
          return true;
      }
    }
  }
  return false;
};
