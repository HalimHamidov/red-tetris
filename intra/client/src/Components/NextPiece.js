import React from "react";
import { Tetrominos } from "../tetrominos";
import { StyledNextDisplay, StyledTitle, StyleStage } from "./styling/StyledDisplay";
import {StyledCell} from "./styling/StyledCell"
const NextPiece = ({ stage, nextPiece }) => (
  <StyledNextDisplay>
    <StyledTitle>Next</StyledTitle>
    <StyleStage>
      {stage && nextPiece.length > 0 && stage?.map((row, i) =>
        row.map((cell, x) => <StyledCell key={x} type={cell[0]} color={Tetrominos[cell[0]]?.color} />)
      )}
    </StyleStage>
  </StyledNextDisplay>
);
export default NextPiece;
