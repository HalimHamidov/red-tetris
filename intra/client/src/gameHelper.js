import { Tetrominos } from "./tetrominos";
export const S_WIDTH = 10;
export const S_HEIGHT = 20;


export const Createstage = (h = S_HEIGHT, w = S_WIDTH) =>
    Array.from(Array(h), () => new Array(w).fill([0, "clear"]));

export const checkcollision = (player, stage, { x: moveX, y: moveY }) => {
    // Использование цикла for для возможности возврата (и прерывания). Невозможно с forEach
    for (let y = 0; y < player.tetromino.length; y++) {
        for (let x = 0; x < player.tetromino.length; x++) {
            // 1. Проверяем, что мы находимся в текущей ячейке Тетромино
            if (player.tetromino[y][x] !== 0) {
                // 2. Проверяем, что наш ход находится внутри высоты игровых площадок (y)
                // Что мы не проходим низ игровой зоны
                if (!stage[y + player.pos.y + moveY] ||
                    // 3. Проверяем, что наш ход находится внутри ширины игровых зон (x)
                    !stage[y + player.pos.y + moveY][x + player.pos.x + moveX] ||
                    // 4. Проверяем, что ячейка, в которую мы двигаемся, не установлена ​​на очистку(clear)
                    stage[y + player.pos.y + moveY][x + player.pos.x + moveX][1] !== "clear"
                )
                    return true;
            }
        }
    }
    // 5. Если все вышеперечисленное неверно
    return false;
};