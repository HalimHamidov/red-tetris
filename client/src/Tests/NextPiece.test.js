import renderer from "react-test-renderer";
import NextPiece from "../Components/NextPiece";
import { Createstage } from "../gameHelper";
let stage = Createstage(4, 4)
test("NextPiece render test ", () => {
    const tree = renderer.create( < NextPiece / > ).toJSON();
    expect(tree).toMatchSnapshot();
});
test("NextPiece render T ", () => {
    const tree = renderer.create( < NextPiece stage = { stage }
        nextPiece = "T" / > ).toJSON();
    expect(tree).toMatchSnapshot();
});