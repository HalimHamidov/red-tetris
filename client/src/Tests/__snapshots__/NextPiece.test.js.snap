// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextPiece render T  1`] = `
<div
  className="sc-gsDKAQ grLxMy"
>
  <div
    className="sc-dkPtRN fdlklY"
  >
    Next
  </div>
  <div
    className="sc-eCImPb hUESGB"
  >
    <div
      className="sc-jRQBWg eVzpdC"
      color="0, 0, 0"
      type={0}
    />
    <div
      className="sc-jRQBWg eVzpdC"
      color="0, 0, 0"
      type={0}
    />
    <div
      className="sc-jRQBWg eVzpdC"
      color="0, 0, 0"
      type={0}
    />
    <div
      className="sc-jRQBWg eVzpdC"
      color="0, 0, 0"
      type={0}
    />
    <div
      className="sc-jRQBWg eVzpdC"
      color="0, 0, 0"
      type={0}
    />
    <div
      className="sc-jRQBWg eVzpdC"
      color="0, 0, 0"
      type={0}
    />
    <div
      className="sc-jRQBWg eVzpdC"
      color="0, 0, 0"
      type={0}
    />
    <div
      className="sc-jRQBWg eVzpdC"
      color="0, 0, 0"
      type={0}
    />
    <div
      className="sc-jRQBWg eVzpdC"
      color="0, 0, 0"
      type={0}
    />
    <div
      className="sc-jRQBWg eVzpdC"
      color="0, 0, 0"
      type={0}
    />
    <div
      className="sc-jRQBWg eVzpdC"
      color="0, 0, 0"
      type={0}
    />
    <div
      className="sc-jRQBWg eVzpdC"
      color="0, 0, 0"
      type={0}
    />
    <div
      className="sc-jRQBWg eVzpdC"
      color="0, 0, 0"
      type={0}
    />
    <div
      className="sc-jRQBWg eVzpdC"
      color="0, 0, 0"
      type={0}
    />
    <div
      className="sc-jRQBWg eVzpdC"
      color="0, 0, 0"
      type={0}
    />
    <div
      className="sc-jRQBWg eVzpdC"
      color="0, 0, 0"
      type={0}
    />
  </div>
</div>
`;

exports[`NextPiece render test  1`] = `
<div
  className="sc-gsDKAQ grLxMy"
>
  <div
    className="sc-dkPtRN fdlklY"
  >
    Next
  </div>
  <div
    className="sc-eCImPb hUESGB"
  />
</div>
`;
